fix: updated the fix for access related issues in dashboard component

Maha14-ui opened the pull request:
closes : https://jsw.ibm.com/browse/MAXUIF-3195?filter=-1 

**Summary**

- here I have updated the fix for access related issues for dashboard value component.
- issues : 
- Content is not within a landmark element
- Accessible name does not match or contain the visible label text
components from carbon-addon-iot-react mentioned below.
 1) valueCard component
2) gaugeCard

**Change List (commits, features, bugs, etc)**

- items_here

**Acceptance Test (how to verify the PR)**

- tests here
Before : 
<img width="1684" height="1038" alt="Screenshot 2025-12-03 at 4 24 37 AM" src="https://github.com/user-attachments/assets/66184930-b45a-4f96-af54-bbe645a8cd79" />
<img width="1679" height="1001" alt="Screenshot 2025-12-03 at 4 25 30 AM" src="https://github.com/user-attachments/assets/a7228ab1-0bf0-4ea2-98c6-fa7d44d7c337" />
<img width="1676" height="998" alt="Screenshot 2025-12-03 at 4 25 40 AM" src="https://github.com/user-attachments/assets/b63990fb-d1e4-43d6-acef-789c2f4c9db6" />
After : 
<img width="1679" height="1008" alt="Screenshot 2025-12-03 at 12 33 44 AM" src="https://github.com/user-attachments/assets/9e6391f2-2ab4-4581-8a28-d85c5c93588c" />
<img width="1676" height="1001" alt="Screenshot 2025-12-03 at 6 19 33 PM" src="https://github.com/user-attachments/assets/35c3b1e9-78f7-45a1-862f-b8dac93228fd" />
<img width="1644" height="1034" alt="Screenshot 2025-12-03 at 6 20 09 PM" src="https://github.com/user-attachments/assets/34191e67-01a9-466d-9db0-6de0fc892bd4" />

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

<!-- For reviewers: do not remove -->

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [x] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [x] PR should link and close out an existing issue

Maha14-ui committed:
fix: updated the fix for access related issues in dashboard component

Maha14-ui committed:
fix: updated the fix for commit failed issue

Maha14-ui committed:
fix: addressed review comments

Maha14-ui committed:
fix: updated the fix for access related issue for gauge card

aqibibm commented on packages/react/src/components/ValueCard/ValueContent.jsx:
```diff
@@ -57,47 +57,51 @@ const ValueContent = ({
-              attribute.secondaryValue && {
-                ...attribute.secondaryValue,
-                value: determineValue(attribute.secondaryValue.dataSourceId, values),
+    <section id="main-content" aria-label="value content">
```
Can't we use same div to give aria-label instead of having <section>

Maha14-ui replied:
hi @aqibibm , I have tried to repeat the issues from reference-app. issue (Content is not within a landmark element) was not repeating and main tag and role is already available from main page so these changes are not required. I have reverted these changes. and updated the PR. please review. Thank you!.

aqibibm commented on packages/react/src/components/GaugeCard/GaugeCard.jsx:
```diff
@@ -93,93 +93,96 @@ const GaugeCard = ({
-                  />
-                  <text
-                    id="gauge-label"
+    <section id="main-content" aria-label="region">
```
Why do we need this?

Maha14-ui replied:
hi @aqibibm ,as I said above. I have reverted these changes. and updated the PR. please review. Thank you!.

aqibibm replied:
I can still see its wrapped in `section` tag. The question is do we really need this `section` tag and why? @Maha14-ui

Maha14-ui replied:
hi @aqibibm , yes section is not required now I have updated the fix without section. its working fine. could you please review it once.
changes in the PR : 1) section removed in value card and gauge card
2) and index.jsx changes for fix

Maha14-ui committed:
Merge branch 'next' into fix-MAXUIF-3195

Maha14-ui committed:
fix: updated the fix for access related issue for dashboad component

Maha14-ui committed:
fix: removed unused variables

Maha14-ui commented:
hi @aqibibm and Team , I was able to reproduce the issue “Accessible name does not match or contain the visible label text.”
I have fixed it in the index.js file and verified the fix in the reference-app through Graphite using my local PAL changes.
updated test ss below. please review the latest changes. Thank you!
Before : 
--------
<img width="1685" height="1033" alt="Screenshot 2025-12-09 at 5 17 12 PM" src="https://github.com/user-attachments/assets/d721f5f7-9c96-458e-bef6-bfbe89194085" />
After : 
------
<img width="1682" height="1038" alt="Screenshot 2025-12-10 at 6 28 36 PM" src="https://github.com/user-attachments/assets/42b401ac-e7b4-4d3a-b7ce-a6ef83df2425" />
<img width="1681" height="1035" alt="Screenshot 2025-12-10 at 6 28 43 PM" src="https://github.com/user-attachments/assets/e893a0b7-de12-4157-93e8-04e490445a51" />

Maha14-ui committed:
fix: updated the format to resolve yarn lint issue

Maha14-ui committed:
fix: removed section and updated the fix

Maha14-ui committed:
fix: reverted removed code
